internal/pkg/api/metrics.go

```diff
@@ -37,83 +43,149 @@ var (
 	cntUploadChunk routeStats
 	cntUploadEnd   routeStats
 	cntArtifacts   artifactStats
+
+	infoReg sync.Once
 )
 
-func InitMetrics(ctx context.Context, cfg *config.Config, bi build.Info) (*api.Server, error) {
-	registry := monitoring.GetNamespace("info").GetRegistry()
-	if registry.Get("version") == nil {
-		monitoring.NewString(registry, "version").Set(bi.Version)
-	}
-	if registry.Get("name") == nil {
-		monitoring.NewString(registry, "name").Set(build.ServiceName)
+// init initializes all metrics that fleet-server collects
+// metrics must be explicitly exposed with a call to InitMetrics
+// FIXME we have global metrics but an internal and external API; this may lead to some confusion.
```

Comment:
Is this something we should have a follow-up issue for? 

What would be your suggestion for an approach to remedy this confusion?

Reply:
it would need a follow up; we can try to make a metrics registry per listener in order to avoid this